File: LeetcodeProblems/blind-75/array/product-of-array-except-self/solution.js
// Product of Array Except Self - LeetCode #238

/**
 * @param {number[]} nums
 * @return {number[]}
 */
var productExceptSelf = function(nums) {
    const n = nums.length;
    const result = new Array(n);
    
    // Left products
    let left = 1;
    for (let i = 0; i < n; i++) {
        result[i] = left;
        left *= nums[i];
    }
    
    // Right products
    let right = 1;
    for (let i = n - 1; i >= 0; i--) {
        result[i] *= right;
        right *= nums[i];
    }
    
    return result;
};

File: LeetcodeProblems/blind-75/array/product-of-array-except-self/solution.py
# Product of Array Except Self - LeetCode #238

from typing import List

class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n = len(nums)
        result = [0] * n
        
        # Left products
        left = 1
        for i in range(n):
            result[i] = left
            left *= nums[i]
        
        # Right products
        right = 1
        for i in range(n-1, -1, -1):
            result[i] *= right
            right *= nums[i]
        
        return result
